src/include/txn_inline.h

```diff
@@ -1061,8 +1061,13 @@ __wt_txn_begin(WT_SESSION_IMPL *session, const char *cfg[])
 
     WT_RET(__wt_txn_config(session, cfg));
 
-    /* Allocate a snapshot if required. */
-    if (txn->isolation == WT_ISO_SNAPSHOT) {
+    /*
+     * Allocate a snapshot if required or update the existing snapshot for non autocommit
+     * transaction. As there is no need of updating snapshot for autocommit transactions as they are
```

Comment:
Suggested rewording:
Allocate a snapshot if required or update the existing snapshot. Do not update the existing snapshot
of autocommit transactions because they are committed at the end of the operation.

Reply:
Changed as per your suggestion.